### PR TITLE
feat(gtest): proper charging of gas allowance in gtest

### PR DESCRIPTION
**Pull request by playX18**
Also added TaskWeights and mailbox cost/threshold into weight dump.

### CODE REVIEW

File: gtest/src/manager/expend.rs [hunk 1]
```diff
@@ -47,18 +47,28 @@ impl ExtManager {
...
+
     pub(crate) fn cost_by_storage_type(storage_type: StorageType) -> u64 {
         // Cost per block based on the storage used for holding
         let schedule = Schedule::default();
         let RentWeights {
             waitlist,
             dispatch_stash,
             reservation,
+            mailbox,
+            mailbox_threshold: _,
```
Comment (StackOverflowExcept1on) [inline gtest/src/manager/expend.rs:66]:
```suggestion
            ..
```

File: gtest/src/manager/block_exec.rs [hunk 2]
```diff
@@ -137,10 +139,108 @@ impl ExtManager {
...
+            }
+
+            if let Some(task) = last_task {
+                stopped_at = Some(bn);
+
+                self.gas_allowance = self
+                    .gas_allowance
+                    .saturating_add(Gas(db_weights.write.ref_time));
+
+                self.task_pool.add(bn, task.clone()).unwrap_or_else(|e| {
```
Comment (techraed) [inline gtest/src/manager/block_exec.rs:218]: Seems no need for this
```suggestion
                self.task_pool.add(bn, task).unwrap_or_else(|e| {
```

File: gtest/src/manager/block_exec.rs [hunk 3]
```diff
@@ -441,11 +541,36 @@ impl ExtManager {
...
+
+fn get_maximum_task_gas(task: &ScheduledTask<ProgramId>) -> Gas {
+    use ScheduledTask::*;
+    let weights = TaskWeights::default();
+    match task {
+        PauseProgram(_) => Gas(0),
+        #[allow(deprecated)]
+        RemoveResumeSession(_) => Gas(0),
+
+        RemoveFromMailbox(_, _) => Gas(weights.remove_from_mailbox.ref_time),
```
Comment (techraed) [inline gtest/src/manager/block_exec.rs:561]: These weights must be used in `manager::task` too.
Now `GearCommonGas:::MIN` is returned

File: gtest/src/manager/task.rs [hunk 4]
```diff
@@ -134,7 +160,7 @@ impl TaskHandler<ProgramId> for ExtManager {
             self.dispatches.push_back(dispatch);
         }
 
-        GearCommonGas::MIN
+        TaskWeights::default().wake_message.ref_time
```
Comment (techraed) [inline gtest/src/manager/task.rs:163]: what about `tasks_wake_message_no_wake` on `None`?

**Review by techraed — APPROVED** [review]

**Pull request merged**